Entity/Components/StatComponent.gd

```diff
@@ -9,5 +9,9 @@ class_name StatComponent
 var _stats: EntityStats = EntityStats.new()
 
 
+func set_stats(new_stats: EntityStats) -> void:
```

Comment:
I found out about the setter-getter syntax for variables after writing this, but it could be nice to implement here
Cf [this link](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_basics.html#properties-setters-and-getters)

Reply:
Now that I'm looking at this I don't think it makes much sense. Because the way we call the getter would be by doing `get_stat_component()._stats` right? I think the explicit `get_stats` is more clear. Unless I misunderstand this syntax

Reply:
doing `get_stat_component()._stats` basically does `get_stat_component().get_stats()`
I find it pretty clear